tilavarauspalvelu/models/reservation/actions.py

```diff
@@ -43,14 +43,57 @@ def get_actual_after_buffer(self) -> datetime.timedelta:
         return buffer_time_after
 
     def to_ical(self) -> bytes:
-        site_name = settings.EMAIL_VARAAMO_EXT_LINK
         language: Lang = (  # type: ignore[assignment]
             self.reservation.reservee_language
             or (self.reservation.user is not None and self.reservation.user.get_preferred_language())
             or settings.LANGUAGE_CODE
         )
 
+        cal = Calendar()
+        cal.add(CalendarProperty.VERSION, "2.0")
+        cal.add(CalendarProperty.PRODID, "-//Helsinki City//NONSGML Varaamo//FI")
+
+        cal.add_component(self._get_ical_timezone())
+        cal.add_component(self._get_ical_event(language=language))
+
+        return cal.to_ical()
+
+    def _get_ical_timezone(self) -> Timezone:
+        """Adds timezone information to the ical calendar event."""
+        timezone = Timezone()
+        timezone.add(TimezoneProperty.TZID, settings.TIME_ZONE)
+
+        # Taken from outlook generated iCal files
+        standard_start = datetime.datetime(1601, 10, 28, 4, 0, 0)
+        daylight_start = datetime.datetime(1601, 3, 25, 3, 0, 0)
+
+        standard = TimezoneStandard()
+        standard.add(name=TimezoneRuleProperty.DTSTART, value=standard_start)
+        standard.add(
+            name=TimezoneRuleProperty.RRULE,
+            value={"FREQ": "YEARLY", "BYDAY": "-1SU", "BYMONTH": 10},
+        )
+        standard.add(name=TimezoneRuleProperty.TZOFFSETFROM, value=datetime.timedelta(hours=3))
+        standard.add(name=TimezoneRuleProperty.TZOFFSETTO, value=datetime.timedelta(hours=2))
+
+        daylight = TimezoneDaylight()
+        daylight.add(name=TimezoneRuleProperty.DTSTART, value=daylight_start)
+        daylight.add(
+            name=TimezoneRuleProperty.RRULE,
+            value={"FREQ": "YEARLY", "BYDAY": "-1SU", "BYMONTH": 3},
+        )
+        daylight.add(name=TimezoneRuleProperty.TZOFFSETFROM, value=datetime.timedelta(hours=2))
+        daylight.add(name=TimezoneRuleProperty.TZOFFSETTO, value=datetime.timedelta(hours=3))
+
+        timezone.add_component(standard)
```

Comment:
I guess strictly speaking these components should only be added if TZID is "Europe/*" -- just in case at some future point the ical files are being generated in UTC but nobody touches the VTimezone part which would then be inappropriate.
(There are maybe ways to programmatically deduce if a given timezone observes DST and with which transition points but that seems like over-complicating things... inasmuch as in practice it will hardly be a non-European timezone, and even if it were, the DST rules might be different anyhow; would need a library and extensive units tests in that case to handle the logic.)

Reply:
The calendar files are always generated in the Helsinki timezone, even if the time saved to database would be in some other timezone (although it's always converted in UTC). I don't think we need to consider this, since this is just for Helsinki City.